File: 875. Koko Eating Bananas/Solution.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left, right, n = 1, 10 ** 9, len(piles)
        while left < right:
            mid, count = (left + right) // 2, 0
            for p in piles:
                count = count + p // mid if p % mid == 0 else count + 1 + p // mid
            if count > h:
                left = mid + 1
            else:
                right = mid
        return left

File: 875. Koko Eating Bananas/Solution.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int left = 1, right = 1000000000, n = piles.length;
        while (left < right) {
            int mid = (left + right) / 2, count = 0;
            for (int i = 0; i < n; i++) {
                count = (piles[i] % mid == 0) ? count + piles[i] / mid : count + 1 + piles[i] / mid;
            }
            if (count > h) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return left;
    }
}